backend/substrapp/ledger_utils.py

```diff
@@ -271,12 +273,20 @@ def _call_ledger(call_type, fcn, args=None, kwargs=None):
         except json.decoder.JSONDecodeError:
             raise LedgerInvalidResponse(response)
 
+        # Set the ledger inside timer as a metric
+        if '__metrics__' in response:
+            metrics = response.pop('__metrics__', None)
+            duration = metrics.get('duration', None)
```

Comment:
Will raise if metrics is `None`

Reply:
You're right about that. In the current chaincode implementation, I think it can't be and present and `None` but it's still an unsafe code.